### ISSUE TITLE
Duplicate Labels Removed

### ISSUE THREAD

**Issue by varun-sappa** [ultralytics/yolov5#12176]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Training

### Bug

train: WARNING ⚠️ /home/varun/Downloads/FLIR_ADAS_v2/images_thermal_train_yolo/images/img9520.jpg: 9 duplicate labels removed
train: WARNING ⚠️ /home/varun/Downloads/FLIR_ADAS_v2/images_thermal_train_yolo/images/img9521.jpg: 40 duplicate labels removed
train: WARNING ⚠️ /home/varun/Downloads/FLIR_ADAS_v2/images_thermal_train_yolo/images/img9522.jpg: 4 duplicate labels removed
train: WARNING ⚠️ /home/varun/Downloads/FLIR_ADAS_v2/images_thermal_train_yolo/images/img9523.jpg: 8 duplicate labels removed
train: WARNING ⚠️ /home/varun/Downloads/FLIR_ADAS_v2/images_thermal_train_yolo/images/img9524.jpg: 7 duplicate labels removed
train: WARNING ⚠️ /home/varun/Downloads/FLIR_ADAS_v2/images_thermal_train_yolo/images/img9525.jpg: 15 duplicate labels removed
train: WARNING ⚠️ /home/varun/Downloads/FLIR_ADAS_v2/images_thermal_train_yolo/images/img9526.jpg: 5 duplicate labels removed
train: WARNING ⚠️ /home/varun/Downloads/FLIR_ADAS_v2/images_thermal_train_yolo/images/img9527.jpg: 15 duplicate labels removed
train: WARNING ⚠️ /home/varun/Downloads/FLIR_ADAS_v2/images_thermal_train_yolo/images/img9528.jpg: 30 duplicate labels removed

This is just a short part of the output. 

this is the .txt file of img9520
0 0.485938 0.492188 0.028125 0.093750
0 0.456250 0.491211 0.028125 0.091797
0 0.268750 0.461914 0.009375 0.041016
2 0.413281 0.475586 0.051562 0.044922
2 0.396094 0.462891 0.039062 0.046875
2 0.367969 0.457031 0.020313 0.011719
2 0.350000 0.467773 0.025000 0.025391
2 0.247656 0.458984 0.035937 0.046875
0 0.128125 0.456055 0.018750 0.052734
0 0.485938 0.492188 0.028125 0.093750
0 0.456250 0.491211 0.028125 0.091797
0 0.268750 0.461914 0.009375 0.041016
2 0.413281 0.475586 0.051562 0.044922
2 0.396094 0.462891 0.039062 0.046875
2 0.367969 0.457031 0.020313 0.011719
2 0.350000 0.467773 0.025000 0.025391
2 0.247656 0.458984 0.035937 0.046875
0 0.128125 0.456055 0.018750 0.052734

### Environment

python 3.8.10
Ubuntu 20.04

### Minimal Reproducible Example

python3 yolov5/train.py --img 640 --batch 1024 --epochs 1  --data coco.yaml --weights yolov5s.pt

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@varun-sappa duplicate Labels Removed

This warning message is generated during training when YOLOv5 detects and removes duplicate labels from the provided dataset. Duplicate labels occur when there are multiple annotations for the same object in an image.

In your case, the `train.py` script has removed duplicate labels for several images, as indicated by the warning messages in your example. The warnings specify the number of duplicate labels removed for each image.

To avoid duplicate labels, ensure that each object in an image is annotated only once. In your provided `.txt` file, there are multiple duplicate annotations for the same objects (`0` and `2`) in the image `img9520`.

Please review your dataset annotations and ensure that each object is annotated only once per image.

Thank you for your contribution! If you would like to submit a pull request (PR) to help improve YOLOv5, please follow the contribution guidelines outlined in the repository.

Let me know if you have any further questions or need additional assistance!

**Comment by varun-sappa** [comment]
Thank You! Found out my mistake. Apoolgies for the inconvenience.

**Comment by glenn-jocher** [comment]
@varun-sappa thank you for your understanding! We appreciate your acknowledgement of the mistake and apologize for any inconvenience it may have caused. If you have any further questions or need any assistance, feel free to ask. We're here to help!

**Comment by Namanm1994** [comment]
@glenn-jocher 
I have a question regarding this. when yolo removes the duplicates from an annotation file, does it remove all the annotation for a particular instance and discards the instance from the training or does it removes `n-1` duplicates to preserve one set of annotation.
Hope my question made sense.
Pleasse let me know, Thank You

**Comment by glenn-jocher** [comment]
@Namanm1994, when YOLOv5 detects duplicate annotations for a particular instance, it removes only the `n-1` duplicates, preserving a single set of annotations for that instance. This ensures that each unique object in the image is still represented once in the training data. The goal is to maintain a clean dataset without multiple identical annotations for the same object, which could otherwise lead to training issues. If you have any more questions, feel free to ask! 😊

**Comment by Namanm1994** [comment]
First of all, Thank You for the quick response, I appreciate it a lot,
And recenty I have been going through the nits and grits of this implementation, although we can still improve a lot, but kudos to you guys for implementing keeping such minute details in mind and managing risks as much as possible.
Highly appreiciated !!
just a follow-up, the `n-1` duplicates are removed by algorithm such as wbf or is it some other logic?

**Comment by glenn-jocher** [comment]
Thank you for the kind words, @Namanm1994! The credit goes to the entire YOLOv5 community and the Ultralytics team for their continuous efforts to improve the project.

Regarding the removal of `n-1` duplicates, it's not done using Weighted Boxes Fusion (WBF) or any complex algorithm. The process is much simpler: during the data loading phase, YOLOv5 checks for overlapping annotations with identical class and coordinates. If duplicates are found, it keeps one annotation and removes the rest. This is a straightforward deduplication step to ensure data cleanliness before training begins.

If you have any more questions or insights, feel free to share. Your engagement with the project is highly valued! 🚀
